src/wasm/wasm-binary.cpp

```diff
@@ -5372,6 +5375,27 @@ bool WasmBinaryBuilder::maybeVisitSIMDLoadStoreLane(Expression*& out,
   return true;
 }
 
+bool WasmBinaryBuilder::maybeVisitPrefetch(Expression*& out, uint32_t code) {
+  PrefetchOp op;
+  switch (code) {
+    case BinaryConsts::PrefetchT:
+      op = PrefetchTemporal;
+      break;
+    case BinaryConsts::PrefetchNT:
+      op = PrefetchNontemporal;
+      break;
+    default:
+      return false;
+  }
+  auto* curr = allocator.alloc<Prefetch>();
```

Comment:
I think it's better to use `builder.makePrefetch`, that is, to leave the direct allocation to the Builder as much as possible, both here an in the s-parser code.